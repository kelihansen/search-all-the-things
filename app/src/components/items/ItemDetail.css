:local(.item-detail) {
  color: #333;  
  background-color: white;
  padding: 2rem;
  margin: 2rem auto;
  width: fit-content;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  h3 {
    margin-top: 1rem;
    width: 500px;    
  }
  span {
    font-style: oblique;
  }
  p {
    margin-top: 0.7rem;
    width: 500px;
  }
  a {
    margin-top: 1rem;
    margin-bottom: 0;
    display: block;
    align-self: flex-end;
    color: #777;
    cursor: pointer;
  }
}