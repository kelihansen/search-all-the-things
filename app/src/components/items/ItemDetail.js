import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { getItemById } from '../../services/api';
import styles from './ItemDetail.css';

export default class ItemDetail extends Component {
  static propTypes = {
    objectID: PropTypes.string.isRequired,
    history: PropTypes.object.isRequired
  };

  state = {
    item: null
  };

  componentDidMount() {
    getItemById(this.props.objectID)
      .then(item => this.setState({ item }));
  }

  handleBack = event => {
    event.preventDefault();
    event.stopPropagation();
    this.props.history.goBack();
  };

  render() {
    const { item } = this.state;

    if(item === null) return null;

    const { largeImageUrl, title, medium, description } = item;

    return (
      <article className={styles['item-detail']}>
        <div>
          <img src={largeImageUrl} alt={title}/>
          <h3>{title}</h3>
          <p><span>{medium}</span></p>
          <p>{description}</p>
        </div>
        <a onClick={this.handleBack}><span>back to search</span></a>
      </article>
    );
  }
}