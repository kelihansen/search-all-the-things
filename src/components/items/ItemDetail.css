:local(.item-detail) {
  color: #333;  
}