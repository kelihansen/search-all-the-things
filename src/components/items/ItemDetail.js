import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { getItem } from '../../services/ch-api';
import styles from './ItemDetail.css';

export default class ItemDetail extends Component {
  static propTypes = {
    objectID: PropTypes.string.isRequired,
    history: PropTypes.object.isRequired
  };

  state = {
    item: null
  };

  componentDidMount() {
    getItem(this.props.objectID)
      .then(item => this.setState({ item }));
  }

  handleBack = event => {
    event.preventDefault();
    event.stopPropagation();
    this.props.history.goBack();
  };

  render() {
    const { item } = this.state;

    if(item === null) return null;

    const { images, title, medium, description } = item.object;

    return (
      <article className={styles['item-detail']}>
        <div>
          <img src={images[0].z.url} alt={title}/>
          <h3>{title}</h3>
          <p><span>{medium}</span></p>
          <p>{description}</p>
        </div>
        <a onClick={this.handleBack}><span>back to search</span></a>
      </article>
    );
  }
}