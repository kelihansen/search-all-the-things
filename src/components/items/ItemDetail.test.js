import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
jest.mock('../../services/ch-api.js');
import ItemDetail from './ItemDetail';

describe('ItemDetail', () => {
  let wrapper;

  beforeAll(() => {
    const mockHistory = { match: { params: { id: 1 } } };
    wrapper = shallow(<ItemDetail objectID="" history={mockHistory}/>);
  });

  it('renders as designed', () => {
    wrapper.setProps({ id: '12345678' });
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('has an image, a heading, two paragraphs, and a link', () => {
    expect(wrapper.find('img')).toHaveLength(1);
    expect(wrapper.find('h3')).toHaveLength(1);
    expect(wrapper.find('p')).toHaveLength(2);
    expect(wrapper.find('a')).toHaveLength(1);
  });

});
