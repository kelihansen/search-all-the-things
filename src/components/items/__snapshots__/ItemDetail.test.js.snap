// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetail renders as designed 1`] = `
<article>
  <div>
    <img
      alt="Cool Museum Thing"
      src="http://coolthings.com.jpg"
    />
    <h3>
      Cool Museum Thing
    </h3>
    <p>
      <span>
        wool, steel
      </span>
    </p>
    <p />
  </div>
  <a
    onClick={[Function]}
  >
    <span>
      back to search
    </span>
  </a>
</article>
`;
